src/styles/main.scss

```diff
@@ -0,0 +1,73 @@
+$day-size: 100px;
+$coloms: 7;
+
+body {
+  margin: 0;
+}
+
+.container {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
+  margin: 0;
+}
+
+.calendar {
+  display: flex;
+  gap: 1px;
+  flex-wrap: wrap;
+
+  max-width: ($day-size * $coloms) + 10;
+}
+
+.calendar__day {
+  display: block;
+  box-sizing: border-box;
+
+  border: solid 1px black;
+  background-color: #eee;
+  line-height: $day-size;
+  text-align: center;
+  font-size: 30px;
+
+  height: $day-size;
+  width: $day-size;
+
+  transition: all 0.5s;
+
+  &:hover {
+    transform: translateY(-20px);
+    background-color: #ffbfcb;
+  }
+}
+
+@for $month-length from 28 to 31 {
+  .calendar--month-length-#{$month-length}
+    .calendar__day:nth-child(n + #{$month-length + 1}) {
+    display: none;
+  }
+}
+
+@for $day-number from 1 through 31 {
+  .calendar__day:nth-child(#{$day-number})::before {
+    content: '#{$day-number}';
+    font-family: Arial, Helvetica, sans-serif;
```

Comment:
```suggestion

```
Move this style to a general class instead of generating the same style for every 31 element